src/recording/recordingmanager.cpp

```diff
@@ -145,23 +167,37 @@ QString& RecordingManager::getRecordingDir() {
 }
 
 // Only called when recording is active.
-void RecordingManager::slotDurationRecorded(QString durationStr)
+void RecordingManager::slotDurationRecorded(quint64 duration)
 {
-    if(m_durationRecorded != durationStr)
+    if(m_secondsRecordedSplit != duration)
     {
-        m_durationRecorded = durationStr;
-        emit(durationRecorded(m_durationRecorded));
+        m_secondsRecordedSplit = duration;
+        if(duration >= m_split_time)
+        {
+            qDebug() << "Splitting after " << duration << " seconds";
+            // Dont generate a new filename.
+            // This will reuse the previous filename but append a suffix.
+            startRecording(false);
+        }
+        emit(durationRecorded(getRecordedDurationStr(m_secondsRecorded+m_secondsRecordedSplit)));
     }
 }
+// Copy from the implementation in enginerecord.cpp
+QString RecordingManager::getRecordedDurationStr(quint64 duration) {
+    return QString("%1:%2")
+                 .arg(duration / 60, 2, 'f', 0, '0')   // minutes
+                 .arg(duration % 60, 2, 'f', 0, '0');  // seconds
+}
 
 // Only called when recording is active.
 void RecordingManager::slotBytesRecorded(int bytes)
 {
     // auto conversion to long
     m_iNumberOfBytesRecorded += bytes;
-    if(m_iNumberOfBytesRecorded >= m_split_size)
+    m_iNumberOfBytesRecordedSplit += bytes;
+    if(m_iNumberOfBytesRecordedSplit >= m_split_size)
```

Comment:
Yes, I understand it in general. Here we have the issue, that we have to split the file before it is bigger than the limit. It is done by actually using smaller limits than the user has selected. 
This was hard to understand for me in the first place. 

A comment like: 
// The  m_split_size limit is tweaked that the user value is not exceeded in any case 
will help. But using an offset here might be better.